test/test_datasets_download.py

```diff
@@ -194,6 +196,17 @@ def collect_download_configs(dataset_loader, name=None, **kwargs):
     return make_download_configs(urls_and_md5s, name)
 
 
+# This is a workaround since fixtures, such as the built-in tmp_dir, can only be used within a test but not within a
+# parametrization. Thus, we use a single root directory for all datasets and remove it when all download tests are run.
+ROOT = tempfile.mkdtemp()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def root():
```

Comment:
I understand that you are using pytest specific functionalities here but could you elaborate on when this is triggered?

Reply:
The `autouse=True` flag tells `pytest` to trigger this automatically even if it is not explicitly used. The `scope="module"` tells `pytest` to run this once per module. In general, everything before the `yield` is the setup whereas everything after it is the teardown.

Here, `pytest` triggers this fixture the first time any test in this module is run and finishes it when the last test is run.